src/nupic/math/StlIo.hpp

```diff
@@ -551,37 +551,48 @@ inline std::ostream &operator<<(std::ostream &out_stream,
 
 //--------------------------------------------------------------------------------
 // std::map
+// Warning: This will not handle elements containing whitespace.
```

Comment:
can we NTA_ASSERT on this?

Reply:
I am not sure this is actually used anywhere...just noticed this was the case while looking to add something like it for std::deque.   The way to allow whitespace would be to use readline and terminate with \n.  We can keep that in mind if we find we are using this function.